### PR TITLE
Fix data in K64F-RIOT-SPI.elf not marked properly 

### PR DESCRIPTION
**Pull request by bubblepipe**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/dev/CONTRIBUTING.md) to this repository.
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/dev/DEVELOPERS.md#code-style).
- [x] I've documented every `RZ_API` function and struct this PR changes.
- [x] I've added tests that prove my changes are effective (required for changes to `RZ_API`).
- [x] I've updated the [Rizin book](https://github.com/rizinorg/book) with the relevant information (if needed).
- [x] I've used AI tools to generate fully or partially these code changes and I'm sure the changes are not copyrighted by somebody else.

**Detailed description**

Patched corner cases for xref data marking.

**Test plan**

I have added one more entry in `test/db/analysis/arm` 

**Closing issues**

https://github.com/rizinorg/rizin/issues/3641

### CODE REVIEW

**Review by notxvilka — APPROVED** [review]

File: librz/core/canalysis.c [hunk 1]
```diff
@@ -4063,7 +4071,8 @@ static void analysis_mark_xrefs_as_data(RzCore *core) {
 			continue;
 		}
 		RzBinSection *sec = rz_bin_get_section_at(bo, target, true);
-		if (sec && (sec->perm & RZ_PERM_X)) {
+		bool target_in_exec = sec && (sec->perm & RZ_PERM_X);
+		if (target_in_exec) {
```
Comment (wargio) [inline librz/core/canalysis.c]: how about moving this into a function?

**Review by wargio — APPROVED** [review]

**Review by Rot127 — APPROVED** [review]
